pgmax/fg/graph.py

```diff
@@ -45,7 +45,8 @@ class FactorGraph:
     """
 
     variables: Union[
-        Mapping[Hashable, groups.VariableGroup],
+        Mapping[int, groups.VariableGroup],
```

Comment:
Better to relax it to `Mapping[Any, groups.VariableGroup]` than restricting it to `int` and `str`, since there is no reason the names should be just `int` or `str`. I will also try to look into this

Reply:
Changed.